MapboxVisionTests/RecordCoordinatorTests.swift

```diff
@@ -72,6 +70,83 @@ final class RecordCoordinatorTests: XCTestCase {
         XCTAssert(directoryExists(at: recordingPath), "Recording should be saved at \(recordingPath) after recording is stopped")
     }
 
+    private class TestClipRequest {
+        let start: Float
+        let end: Float
+
+        init(start: Float, end: Float) {
+            self.start = start
+            self.end = end
+        }
+    }
+
+    private class RecordDelegate: RecordCoordinatorDelegate {
+        private let onRecordingStopped: () -> Void
+
+        init(onRecordingStopped: @escaping () -> Void) {
+            self.onRecordingStopped = onRecordingStopped
+        }
+
+        func recordingStarted(path: String) {
+        }
+
+        func recordingStopped(recordingPath: RecordingPath) {
+            onRecordingStopped()
+        }
+    }
+
+    private func performVideoClipping(with requests: [TestClipRequest], _ message: String = "", line: UInt = #line) {
+        let expectation = XCTestExpectation(description: "All the requested clips are successfully cut")
+        let videoSource = MockVideoSource()
+        videoSource.add(observer: coordinator)
+        videoSource.start()
+
+        let recordDelegate = RecordDelegate {
+            let path = DocumentsLocation.currentRecording.path
+            let directoryContent = try? FileManager.default.contentsOfDirectory(at: URL(string: path)!, includingPropertiesForKeys: nil, options: [])
+
+            if let directoryContent = directoryContent {
+                for file in directoryContent where file.hasDirectoryPath {
+                    let subDirectoryContent = (try? FileManager.default.contentsOfDirectory(at: file, includingPropertiesForKeys: nil, options: []))
+
+                    if let subDirectoryContent = subDirectoryContent {
+                        let numberOfVideos = subDirectoryContent.filter {
+                            $0.pathExtension == self.videoSettings.fileExtension
+                        }
+                        .count
+
+                        XCTAssert(numberOfVideos == requests.count, message, line: line)
+                    }
+                }
+            }
+            expectation.fulfill()
+        }
+
+        coordinator.delegate = recordDelegate
+        coordinator.startRecording(referenceTime: 0, videoSettings: videoSettings) {}
+        DispatchQueue.main.asyncAfter(deadline: .now() + 4.0) { [unowned self] in
+            requests.forEach {
+                self.coordinator.makeClip(from: $0.start, to: $0.end)
+            }
+
+            DispatchQueue.main.asyncAfter(deadline: .now() + 1.0) { [unowned self] in
+                self.coordinator.stopRecording()
+            }
+        }
+
+        XCTWaiter().wait(for: [expectation], timeout: 10.0)
+    }
+
+    func testVideoClipping() {
```

Comment:
Could you please organize test-cases together and private helper functions separately (later in the file)?

Reply:
moved to the end of the class